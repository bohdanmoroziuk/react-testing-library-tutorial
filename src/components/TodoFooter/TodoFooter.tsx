import { Link } from "react-router-dom"

import 'components/TodoFooter/TodoFooter.css';

export interface TodoFooterProps {
  numberOfIncompleteTodos: number;
}

export default function TodoFooter({ numberOfIncompleteTodos }: TodoFooterProps) {
  return (
    <div className="todo-footer" data-testid="todo-footer">
      <p>{numberOfIncompleteTodos} {numberOfIncompleteTodos === 1 ? "task" : "tasks"} left</p>
      <Link to="/followers">Followers</Link>
    </div>
  );
}
