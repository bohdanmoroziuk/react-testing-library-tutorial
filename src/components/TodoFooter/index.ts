import TodoFooter from 'components/TodoFooter/TodoFooter';
export default TodoFooter;
