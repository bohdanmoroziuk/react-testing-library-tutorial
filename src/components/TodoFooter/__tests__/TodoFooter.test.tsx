import { render, screen } from '@testing-library/react';
import { ReactNode } from 'react';
import { MemoryRouter } from 'react-router-dom';

import TodoFooter from 'components/TodoFooter';

const renderWithRouter = (children: ReactNode) => render(
  <MemoryRouter>
    {children}
  </MemoryRouter>
);

describe('<TodoFooter />', () => {
  test('renders successfully', () => {
    renderWithRouter(
      <TodoFooter numberOfIncompleteTodos={0} />
    );

    const target = screen.getByTestId('todo-footer');

    expect(target).toMatchSnapshot();
  });

  test('renders the correct amount of incomplete tasks', () => {
    renderWithRouter(
      <TodoFooter numberOfIncompleteTodos={5} />
    );

    const target = screen.getByText(/5 tasks left/i);

    expect(target).toBeInTheDocument();
  });

  test('renders "task" when the number of incomplete task is one', () => {
    renderWithRouter(
      <TodoFooter numberOfIncompleteTodos={1} />
    );

    const target = screen.getByText(/1 task left/i);

    expect(target).toBeInTheDocument();
  });
});