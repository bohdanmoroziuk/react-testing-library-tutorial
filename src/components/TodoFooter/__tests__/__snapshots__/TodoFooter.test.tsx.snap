// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoFooter /> renders successfully 1`] = `
<div
  class="todo-footer"
  data-testid="todo-footer"
>
  <p>
    0
     
    tasks
     left
  </p>
  <a
    href="/followers"
  >
    Followers
  </a>
</div>
`;
